src/psyclone/domain/lfric/kernel_interface.py

```diff
@@ -0,0 +1,412 @@
+
+    def stencil(self, arg, var_accesses=None):
+        ''' Not implemented '''
+        raise NotImplementedError("stencil not implemented")
+
+    def operator(self, arg, var_accesses=None):
+        ''' Create LFRic size and operator data objects '''
+
+        op_size = OperatorSizeDataSymbol(
+            "ncell_3d", interface=self.read_access)
+        self._symbol_table.add(op_size)
+        self._arglist.append(op_size)
+        op_arg = OperatorDataSymbol(
+            arg.name, [1, 1, 1],
+            arg.function_space_from.orig_name,
+            arg.function_space_to.orig_name,
+            interface=ArgumentInterface(INTENT_MAPPING[arg.intent]))
+        self._symbol_table.add(op_arg)
+        self._arglist.append(op_arg)
+        self._operators_list.append(op_arg)
+        self._operator_dims_map[op_arg] = op_size
+
+    def cma_operator(self, arg, var_accesses=None):
+        ''' Not implemented '''
+        raise NotImplementedError("cma_operator not implemented")
+
+    def scalar(self, scalar_arg, var_accesses=None):
+        ''' Create LFRic scalar symbol '''
+        mapping = {
+            "integer": LfricIntegerScalarDataSymbol,
+            "real": LfricRealScalarDataSymbol,
+            "logical": LfricLogicalScalarDataSymbol}
+        try:
+            arg = mapping[scalar_arg.intrinsic_type](
+                scalar_arg.name, interface=ArgumentInterface(
+                    INTENT_MAPPING[scalar_arg.intent]))
+        except KeyError:
+            raise NotImplementedError(
+                "scalar of type {0} not implemented"
+                "".format(scalar_arg.intrinsic_type))
+        self._symbol_table.add(arg)
+        self._arglist.append(arg)
+
+    def fs_common(self, function_space, var_accesses=None):
+        ''' Create LFRic Number of Dofs object '''
+
+        fs_name = function_space.orig_name
+        arg = NumberOfDofsDataSymbol("ndf_{0}".format(fs_name), fs_name,
+                                     interface=self.read_access)
+        self._symbol_table.add(arg)
+        self._arglist.append(arg)
+        self._ndofs_map[fs_name] = arg
+
+    def fs_intergrid(self, function_space, var_accesses=None):
+        ''' Not implemented '''
+        raise NotImplementedError("fs_integrated not implemented")
+
+    def fs_compulsory_field(self, function_space, var_accesses=None):
+        '''Create LFRic Number of Unique dofs and dofmap objects'''
+
+        fs_name = function_space.orig_name
+        arg = NumberOfUniqueDofsDataSymbol("undf_{0}".format(fs_name), fs_name,
+                                           interface=self.read_access)
+        self._symbol_table.add(arg)
+        self._arglist.append(arg)
+        self._undfs_map[fs_name] = arg
+
+        # The dimension argument may not have been declared yet so use
+        # an integer as a placeholder.
+        arg = DofMapDataSymbol("dofmap_{0}".format(fs_name), [1], fs_name,
```

Comment:
Again, this implies to me that something isn't quite right with the interface.

Reply:
Agreed. Your suggestion works a treat.